File: C assignments/credit.c
//checks if a credit card number is valid and returns which company it belongs to. usage - ./credit

#include <cs50.h>
#include <stdio.h>
//long calculate_sum1(long number);

int main(void)
{
    long number;
    number = get_long("Number: ");
    long sum1 = 0;
    long sum2 = 0;
    long starting_numbers = number;
    long starting_number = number;

//even digits sum
    for (long i = number; i>1; i = (i - i%10)/10)
    {
        long even_digits;
        even_digits = 2*(((i - i%10)/10)%10);
        long split_ed;
        split_ed = even_digits%10 + ((even_digits - (even_digits%10))/10);
        sum1 = sum1 + split_ed;
        i = (i - i%10)/10;
    }

//odd digits sum
    for (long j = number; j>1; j = (j - j%10)/10)
    {
        long odd_digits = j%10;
        long split_od = odd_digits%10 + ((odd_digits - (odd_digits%10))/10);
        sum2 = sum2 + split_od;
        j = (j - j%10)/10;
    }

//total sum
    int total_sum = sum1 + sum2;

//number of digits
    long num = number;
    int digits = 0;
    while (num>0)
    {
        num = num/10;
        digits++;
    }
//    printf("%i\n", digits);

//starting digits
    long start = number;
    while (start > 100)
    {
        start = (start - start%10)/10;
    }
//    printf("%ld\n", start);

//validation
    if ((digits != 15 && digits != 16 && digits != 13) || total_sum%10 != 0)
    {
        printf("INVALID\n");
    } else if ((start - start%10)/10 == 4)
    {
        printf("VISA\n");
    } else if (start == 34 || start == 37)
    {
        printf("AMEX\n");
    } else if ((start - start%10)/10 == 5 && (start%10 >= 1 && start%10 <= 5))
    {
        printf("MASTERCARD\n");
    } else
    {
        printf("INVALID\n");
    }
}

File: Python Assignments/credit.py
# TODO
def main():
    try:
        number = int(input("Number:"))
    except ValueError:
        number = int(input("Number:"))

    i = number
    o = number
    odd_sum = 0
    even_sum = 0

    #even sum
    while i>1:
        i = (i-i%10)/10
        ed = i%10
        split_ed = (2*ed)%10 + (2*ed - (2*ed)%10)/10
        even_sum += split_ed
        i = (i-i%10)/10

    #odd sum
    while o > 1:
        od = o%10
        odd_sum += od
        o = (o-o%10)/10
        o = (o-o%10)/10

    total_sum = even_sum + odd_sum

    if total_sum%10 == 0.0:
        if (len(str(number)) == 13 and first_digits(number) in range(40, 50)):
            print("VISA")
        elif (len(str(number)) == 16):
            if first_digits(number) in range(40, 50):
                print("VISA")
            elif first_digits(number) in range(51, 56, 1):
                print("MASTERCARD")
            else:
                print("INVALID")
        elif (len(str(number)) == 15 and (first_digits(number) == 37 or first_digits(number) == 34)):
            print("AMEX")
        else:
            print("INVALID")
    else:
        print("INVALID")

def first_digits(start):
    while start >= 100:
        start = (start - start%10)/10
    return start

main()